src/pixelator/graph/backends/implementations/_networkx.py

```diff
@@ -904,18 +936,28 @@ def _prob_edge_weights(
     P_step = P_step.multiply(A)
 
     # Compute bi-directional transition probabilities which are symmetric.
-    # Now we get the probability of going from i to j and back again in k steps,
-    # so it doesn't matter if we start in i or j.
+    # Now we get the probability of going from i to j and back again in k
+    # steps, so it no longer matters if we start in i or j.
     P_step_bidirectional = P_step.multiply(P_step.T)
 
     # Extract the transition probabilities for edges in g
-    edges = list(g.edges)
-    edge_probs = P_step_bidirectional[
-        [list(g.nodes).index(u) for u, _ in edges],
-        [list(g.nodes).index(v) for _, v in edges],
-    ]
+    edges = np.array(list(g.edges))
+    nodes = np.array(list(g.nodes))
```

Comment:
I don't think there is a need to make the `edge`/`nodes` into lists here. In fact I think skipping it should be slightly more performant since it should avoid an extra allocation of the data.